### PR TITLE
Feature activityflow

### PR DESCRIPTION
**Pull request by StefanEvanghelides**
This branch contains the new flow of activities:
- Firstly, it looks for the server (using NDS and zeroconf - if it fails, you can enter the IP manually).
- Then, you attempt to log in.
- ChangeIpDialog was changed to DiscoverServerDialog
- extracted strings and included them in the string.xml

### CODE REVIEW

File: Hestia/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -5,45 +5,38 @@
...
     <string name="standardPass">password</string>
+
     <string name="correctLoginToast">Correct, redirecting."</string>
     <string name="incorrectLoginToast">Invalid credentials.</string>
-    <string name="loginIntentExtra">login</string>
     <string name="saveLogin">savelogin</string>
 
     <string name="hestiaIp">HESTIA.IP</string>
     <string name="ipOfServer">IP_OF_SERVER</string>
     <!-- Menu items text-->
```
Comment (StefanEvanghelides) [inline Hestia/app/src/main/res/values/strings.xml]: I like the splitting.

File: .travis.yml [hunk 2]
```diff
@@ -23,8 +23,7 @@ android:
     - addon-google_apis-google-19
 
     #System images
-    - sys-img-armeabi-v7a-android-22
-    - sys-img-armeabi-v7a-android-17
+    - sys-img-armeabi-v7a-android-16
```
Comment (FrankTN) [inline .travis.yml:26]: This should speed up the build somewhat, version 16 is supposedly the fastest image.

File: Hestia/app/build.gradle [hunk 3]
```diff
@@ -5,7 +5,7 @@ android {
     buildToolsVersion "25.0.3"
     defaultConfig {
         applicationId "com.rugged.application.hestia"
-        minSdkVersion 15
+        minSdkVersion 16
```
Comment (FrankTN) [inline Hestia/app/build.gradle:8]: Minimum sdk changed to 16

File: Hestia/app/src/main/java/hestia/UI/dialogs/AddDeviceDialog.java [hunk 4]
```diff
@@ -1,30 +1,36 @@
...
 import hestia.backend.ServerCollectionsInteractor;
 import hestia.backend.exceptions.ComFaultException;
 import hestia.backend.models.RequiredInfo;
 
 /**
-* This class opens the dialog to enter the collection name and plugin name.
-* It then sends this to the networkHandler which tries to get the required info.
-* If this works it consecutively opens a new dialog for the other info.
-* @see EnterRequiredInfoDialog
+<<<<<<< HEAD
```
Comment (FrankTN) [inline Hestia/app/src/main/java/hestia/UI/dialogs/AddDeviceDialog.java]: This should first be resolved

**Review by FrankTN — CHANGES_REQUESTED** [review]
A very big pull request, looks good apart from a merge conflict with the documentation

File: Hestia/app/build.gradle (hunk 3 above)
Comment (StefanEvanghelides) [reply Hestia/app/build.gradle:8]: good change. Missed that one

File: .travis.yml (hunk 2 above)
Comment (StefanEvanghelides) [reply .travis.yml:26]: Nice

File: Hestia/app/src/main/java/hestia/UI/activities/home/HomeActivity.java [hunk 5]
```diff
@@ -101,7 +101,9 @@ private void setupServerCollectionInteractor() {
      */
     public void refreshUserInterface(){
         DeviceListFragment fragment = (DeviceListFragment) fragmentManager.findFragmentByTag("DeviceListFragment");
-        fragment.populateUI();
+        if(fragment != null){
```
Comment (StefanEvanghelides) [inline Hestia/app/src/main/java/hestia/UI/activities/home/HomeActivity.java]: good one.

**Review by FrankTN — APPROVED** [review]

**Pull request merged**